pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -308,3 +334,62 @@ func createValidImageVersionLabel(image string) string {
 	}
 	return cephImageVersion
 }
+
+func waitForPvcToExpandWithTimeout(ctx context.Context, client client.Client, pvcResizeMap map[string]pvcResize, namespace string, pvcResizeWaitTimeOut time.Duration) {
+	currentTime := time.Now()
+	if pvcResizeWaitTimeOut == 0 {
+		pvcResizeWaitTimeOut = defaultWaitTimeoutForHealthyOSD
+	}
+	timeout := currentTime.Add(pvcResizeWaitTimeOut)
+	allPvcResize := false
+	for !allPvcResize {
+		if time.Now().After(timeout) {
```

Comment:
Always use UTC time so there is no bug around time zones or DST.
```suggestion
		if time.UtcNow().After(timeout) {
```